services/vod-to-live/service/README.md

```diff
@@ -0,0 +1,250 @@
+the protected content and encrypt the newly created live stream , CPIX files are
+created and stored in Azure Storage. Access to the files is granted to the USP
+using shared access signatures (SAS) with time-based limitations.
+
+A new DRM key will be created for each Channel through the Axinom Key Service
+API. The key identifier for the Channel will be sent in a RabbitMQ message of
+the type `LiveStreamProtectionKeyCreatedEvent`. This identifier will be stored
+with the Channel's metadata in Azure Storage.
+
+#### Playlist prolongation
+
+When the Playlist prolongation feature is enabled, the VOD-to-Live Service will
+automatically extend the duration of any playlist that has a duration of less
+than 24 hours, so that it is exactly 24 hours long. This is achieved by using a
+placeholder video to fill the remaining time.
+
+## Setup
+
+- Prerequisites: The
+  [Unified Virtual Channel](https://beta.docs.unified-streaming.com/documentation/virtual-channel/gettingstarted/index.html)
+  should be set up.
+- Follow the instructions from the `README.md` file in the Media Solution root
+  folder.
+- Open the configuration file `.env` for the service and set missing values
+  accordingly.
+
+### Service configuration
+
+#### General settings
+
+`PRE_PUBLISHING_WEBHOOK_SECRET` - This is a mandatory setting. The secret key
+used to sign validation messages between the Managed Channel and VOD-to-Live
+services. It can be retrieved from the Admin portal when setting up the
+Pre-Publishing webhook for the Channel Service.
+
+`CATCH_UP_DURATION_IN_MINUTES` - This setting defaults to 60 minutes and adds
+this duration to the playlist to create smoother transitions between playlists.
+
+`PROLONG_PLAYLIST_TO_24_HOURS` - This feature is enabled by default and prolongs
+the playlist duration to 24 hours.
+
+`CHANNEL_PROCESSING_WAIT_TIME_IN_SECONDS` - By default, this setting is set to
+10 minutes and determines the amount of time the VOD-to-Live service waits
+before the channel goes live.
+
+#### USP Virtual Channel API settings
+
+`VIRTUAL_CHANNEL_MANAGEMENT_API_BASE_URL` - This is the mandatory setting for
+base URL of the Unified Virtual Channel Management API.
+
+`VIRTUAL_CHANNEL_MANAGEMENT_API_KEY` - This is an optional API key used for
+authorization with the Virtual Channel Management API.
+
+`VIRTUAL_CHANNEL_ORIGIN_BASE_URL` - This is the mandatory setting for base URL
+of the Unified Streaming Origin or the CDN for Origin.
+
+#### Azure Storage settings
+
+`AZURE_STORAGE_CONNECTION` - This is the mandatory setting for the connection
+string of the Azure storage.
+
+`AZURE_BLOB_CONTAINER_NAME` - This is the mandatory setting. Name of the
+container in Azure Storage where the service will store channel data.
+
+#### DRM settings
+
+To enable DRM protection of the live stream, fill out the configurations below
+with the values available on the Axinom Portal.
+
+`KEY_SERVICE_API_BASE_URL` - This setting is optional.
```

Comment:
Technically, this is correct, but during development the variable must be commented out to truly be optional, as it will resolve to an empty string and will throw an error on startup if not specified. This is not specific to this case and is more of an env-var issue, which I have described here: https://github.com/evanshortiss/env-var/issues/167
Not sure if there is any decent workaround available for now, in case we want to disable DRM

Reply:
Starting with version `0.32.0-rc.2`, `@axinom/mosaic-service-common` now includes an `asOptionalUrlString` `env-var` extension function. Please apply it to this variable.